JavaFXUserInterface/src/main/java/controllers/AssetTypeController.java

```diff
@@ -111,13 +119,110 @@ public void fillAssetTypeTable() {
+                Text warningAssets = new Text(String.valueOf(assetType.getCountWarning()));
+                Text failedAssets = new Text(String.valueOf(assetType.getCountFailed()));
+
+                assetTypeName.setLayoutX(15.0);
+                assetTypeName.setLayoutY(35.0);
+                okLabel.setLayoutX(15.0);
+                okLabel.setLayoutY(76.0);
+                advisoryLabel.setLayoutX(15.0);
+                advisoryLabel.setLayoutY(119.0);
+                cautionLabel.setLayoutX(15.0);
+                cautionLabel.setLayoutY(162.0);
+                warningLabel.setLayoutX(15.0);
+                warningLabel.setLayoutY(205.0);
+                failedLabel.setLayoutX(15.0);
+                failedLabel.setLayoutY(248.0);
+
+                okBox.setLayoutX(131.0);
+                okBox.setLayoutY(68.0);
+                advisoryBox.setLayoutX(131.0);
+                advisoryBox.setLayoutY(111.0);
+                cautionBox.setLayoutX(131.0);
+                cautionBox.setLayoutY(154.0);
+                warningBox.setLayoutX(131.0);
+                warningBox.setLayoutY(197.0);
+                failedBox.setLayoutX(131.0);
+                failedBox.setLayoutY(240.0);
+
+                okBox.getChildren().add(okAssets);
+                advisoryBox.getChildren().add(advisoryAssets);
+                cautionBox.getChildren().add(cautionAssets);
+                warningBox.getChildren().add(warningAssets);
+                failedBox.getChildren().add(failedAssets);
+
+                pane.getChildren().addAll(assetTypeName, okLabel, advisoryLabel, warningLabel, cautionLabel, failedLabel, okBox, advisoryBox,
+                        warningBox, cautionBox, failedBox);
+                assetsTypeThumbPane.getChildren().add(pane);
+            }
+        }
+    }
+
     /**
      * Adds mouse events to all the buttons and allows columns to be filled with
      * give information
      *
      * @author Shirwa
      */
     public void attachEvents() {
+        // As the window expands or shrinks, asset panes will adjust to the window size accordingly
+        assetsTypeThumbPane.widthProperty().addListener((obs, oldVal, newVal) -> {
+            double assetTypeFlowWidth = (double) newVal - 54;
+            if(assetsTypeThumbPane.getChildren().size() < 6) {
+                int nbOfPanes = assetsTypeThumbPane.getChildren().size();
+                assetTypeFlowWidth = assetTypeFlowWidth - (247 * nbOfPanes);
+                assetTypeFlowWidth = assetTypeFlowWidth / (nbOfPanes - 1);
+                assetsTypeThumbPane.setHgap(assetTypeFlowWidth);
+            } else {
+                int nbOfPanes = (int) (assetTypeFlowWidth / 247);
+                assetTypeFlowWidth = (assetTypeFlowWidth % 247);
+                assetTypeFlowWidth = assetTypeFlowWidth / (nbOfPanes - 1);
+                assetsTypeThumbPane.setHgap(assetTypeFlowWidth);
```

Comment:
Instead of having hardcoded values for integer layout and styling values inlined into the code, maybe it would be a better idea to define those in the class as static final variables that you can call at different places. This will facilitate the maintainability of the app.

Reply:
done